MovieSharing/VideoAPIService.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  VideoRequest.swift
+//  MovieSharing
+//
+//  Created by Mariam Korashi on 29.09.21.
+//
+
+import Foundation
+
+
+
+struct VideoAPIService {
+    
+    // MARK: - Types
+    
+    enum VideoError: Error {
+        case noDataAvailable
+        case canNotProcessData
+    }
+    
+    // MARK: - Properties
+    
+    let resourceURL: URL
+    
+    // MARK: - Public API
+    
+    init() {
+        let resourceString = "https://www.googleapis.com/youtube/v3/playlistItems?part=snippet&playlistId=PLeagipoZmyfnIxkk9qKN-ewkuDeI-JP0i&key=AIzaSyB2WofOa5qg3RLlCnwGJ31Mw9O-PaKJS_8"
+        
+        guard let resourceURL = URL(string: resourceString) else {fatalError()}
```

Comment:
put the fatalError on a new line like
```suggestion
        guard let resourceURL = URL(string: resourceString) else {
        fatalError()
        }
```

but apart from that we usually don't want to include anything that can cause the app to crash.
I would suggest: do not include the resourceURL stuff in the init() but instead in the getVideos, that will allow you an early return in case the URL isn't converted like expected.